server/common/corpora.py

```diff
@@ -0,0 +1,81 @@
+"""
+Corpora schema conventions support.  Helper functions for reading.
+
+https://github.com/chanzuckerberg/corpora-data-portal/blob/main/backend/schema/corpora_schema.md
+
+https://github.com/chanzuckerberg/corpora-data-portal/blob/main/backend/schema/corpora_schema_h5ad_implementation.md
+"""
+import json
+
+from server.cli.upgrade import validate_version_str
+
+
+def corpora_get_versions_from_anndata(adata):
+    """
+    Given an AnnData object, return:
+        * None - if not a Corpora object
+        * [ corpora_schema_version, corpora_encoding_version ] - if a Corpora object
+
+    Implements the identification protocol defined in the spect.
+    """
+
+    # per Corpora AnnData spec, this is a corpora file if the following is true
+    if "version" not in adata.uns or "corpora_schema_version" not in adata.uns["version"]:
+        # oops, not corpora
+        return None
+
+    corpora_schema_version = adata.uns["version"].get("corpora_schema_version", None)
+    corpora_encoding_version = adata.uns["version"].get("corpora_encoding_version", None)
+
+    # TODO: spec says these must be SEMVER values, so check.
+    if validate_version_str(corpora_schema_version) and validate_version_str(corpora_encoding_version):
+        return [corpora_schema_version, corpora_encoding_version]
+
+    return None
+
+
+def corpora_get_props_from_anndata(adata):
+    """
+    Get Corpora dataset properties from an AnnData
+    """
+    versions = corpora_get_versions_from_anndata(adata)
+    if versions is None:
+        return None
+    [corpora_schema_version, corpora_encoding_version] = versions
+    version_is_supported = (
+        corpora_schema_version is not None
+        and corpora_encoding_version is not None
+        and corpora_schema_version.startswith("1.")
+        and corpora_encoding_version.startswith("0.1.")
+    )
+
+    if not version_is_supported:
+        raise ValueError("Unsupported Corpora schema version")
+
+    Required_Simple_Fields = [
```

Comment:
Just a nit: why is the casing weird on these variables?

Reply:
casing?

Reply:
The capitalization of the beginning of each word threw me off :P Haven't seen it before and didn't know what it meant.

Reply:
lolz, no idea - just me trying to imply constants.  I'll remove it on the assumption you are the person who defines code convensions.